govcd/edgegateway_lb_service_monitor.go

```diff
@@ -0,0 +1,187 @@
+	if err := validateCreateLBServiceMonitor(lbMonitorConfig); err != nil {
+		return nil, err
+	}
+
+	httpPath, err := eGW.getProxiedEdgeEndpoint("/loadbalancer/config/monitors")
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+	// We expect to get http.StatusCreated or if not an error of type types.NSXError
+	resp, err := eGW.client.ExecuteRequestHTTPCodeOrTypedError(http.StatusCreated, httpPath, http.MethodPost, "application/xml", "error creating load balancer service monitor: %s", lbMonitorConfig, &types.NSXError{})
+	if err != nil {
+		return nil, err
+	}
+	location := resp.Header.Get("Location")
+
+	// Last element in location header is the service monitor ID
+	// i.e. Location: [/network/edges/edge-3/loadbalancer/config/monitors/monitor-5]
+	if location == "" {
+		return nil, fmt.Errorf("unable to retrieve ID for new load balancer service monitor with name %s", lbMonitorConfig.Name)
+	}
+	splitLocation := strings.Split(location, "/")
+	lbMonitorID := splitLocation[len(splitLocation)-1]
+	readMonitor, err := eGW.ReadLBServiceMonitor(&types.LBMonitor{ID: lbMonitorID})
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve monitor with ID (%s) after creation: %s", readMonitor.ID, err)
+	}
+	return readMonitor, nil
+}
+
+// ReadLBServiceMonitor is able to find the types.LBMonitor type by Name and/or ID.
+// If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
+// name do not match.
+func (eGW *EdgeGateway) ReadLBServiceMonitor(lbMonitorConfig *types.LBMonitor) (*types.LBMonitor, error) {
+	if err := validateReadLBServiceMonitor(lbMonitorConfig); err != nil {
+		return nil, err
+	}
+
+	httpPath, err := eGW.getProxiedEdgeEndpoint("/loadbalancer/config/monitors")
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	// Anonymous struct to unwrap "monitor response"
+	lbMonitorResponse := &struct {
+		LBMonitors []*types.LBMonitor `xml:"monitor"`
+	}{}
+
+	// This query returns all service monitors as the API does not have filtering options
+	_, err = eGW.client.ExecuteRequest(httpPath, http.MethodGet, "application/xml", "unable to read Load Balancer monitor: %s", nil, lbMonitorResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	// Search for monitor by ID or by Name
+	for _, monitor := range lbMonitorResponse.LBMonitors {
+		// If ID was specified for lookup - look for the same ID
+		if lbMonitorConfig.ID != "" && monitor.ID == lbMonitorConfig.ID {
+			return monitor, nil
+		}
+
+		// If Name was specified for lookup - look for the same Name
+		if lbMonitorConfig.Name != "" && monitor.Name == lbMonitorConfig.Name {
+			// We found it by name. Let's verify if search ID was specified and it matches the lookup object
+			if lbMonitorConfig.ID != "" && monitor.ID != lbMonitorConfig.ID {
```

Comment:
Is this not dead code - not accessible?

Reply:
Shouldn't be. The above condition checks for `lbMonitorConfig.ID != "" && monitor.ID `**==** `lbMonitorConfig.ID`

Reply:
Oh I got it, there is difference == and ~= in ID's